src/Rules/Doctrine/ORM/EntityColumnRule.php

```diff
@@ -160,13 +160,14 @@ public function processNode(Node $node, Scope $scope): array
 			return [];
 		}
 
-		$propertyTransformedType = TypeTraverser::map($propertyType, static function (Type $type, callable $traverse): Type {
+		// If the type descriptor does not precise the types inside the array, don't report errors if the field has a more precise type
```

Comment:
I disagree with this. If Doctrine can write any array to a property, it's wrong to expect a narrower type.

Reply:
This comment is about explaining the *existing* logic that you added in https://github.com/phpstan/phpstan-doctrine/commit/e745610aa35228935edf6e7fbf79e02d505c37a6

The change I'm doing here is *disabling* this logic for some cases (when the type descriptor is precise), not enabling it, to avoid reporting errors that are wrong.

I tried to remove this transformation entirely, but this broke some existing tests (the ones added in that commit)

Reply:
Note however that the only type descriptors that return `array` as type (and so would trigger this logic) are `json_array` (deprecated in DBAL 2.10 and removed in DBAL 3.0) and `array` (deprecated in DBAL 3.4+).

So maybe we could drop this logic and return `NeverType` as done in the JsonType descriptor

Reply:
Oh I get it now, it's nice 👍 

I'm not rushing to remove functionality, people use phpstan-doctrine with years-old versions of Doctrine.